server/s3_put_object_action.cc

```diff
@@ -58,7 +58,20 @@ S3PutObjectAction::S3PutObjectAction(
   old_object_oid = {0ULL, 0ULL};
   old_layout_id = -1;
   new_object_oid = {0ULL, 0ULL};
-
+  fault_mode_active = false;
+  last_object_size = 0;
+  primary_object_size = 0;
+  total_object_size_consumed = 0;
+  no_of_blocks_written = 0;
+  create_fragment_when_write_success = false;
+
+  // Default to 10 objects in S3 fault mode
+  max_objects_in_s3_fault_mode = 10;
+  current_fault_iteration = 0;
+  if (S3Option::get_instance()->get_max_objects_in_fault_mode() > 0) {
```

Comment:
should we put WARN or INFO message that the value specified in the config file is not within "unsigned short" range in else part ???

Reply:
Also can CORTX handle unsigned short highest value : 65k ??

If yes, we should ask QA to specifically test with the highest value.

otherwise can we put a check, like if value (in config file) is greater than N, then set the "max_objects_in_s3_fault_mode" to N

Reply:
We would certainly need conditioning in code, and if found greater than N, set it to N. 
Also, practically, we may end up supporting very less number of objects in fault mode, e.g., 3 or 5 or 10 or 30...etc. Moreover, @knrajnambiar76 recommends it to be 1, but I'm not sure.